extensions/terminal-suggest/src/terminalSuggestMain.test.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import assert from 'assert';
+import 'mocha';
+import { availableSpecs, getCompletionItemsFromSpecs } from './terminalSuggestMain';
+
+suite('Terminal Suggest', () => {
+	const availableCommands = new Set(['cd', 'code', 'code-insiders', 'ls', 'pwd']);
+	suiteSetup(async function () {
+
+	});
+	suiteTeardown(async function () {
+	});
+
+	suite('prefix: c', () => {
+		createTestCase(`No available commands:`, 'c|', [], 'neither', availableSpecs, new Set());
+		createTestCase('Available commands', 'c|', ['cd', 'code', 'code-insiders'], 'neither', availableSpecs, availableCommands);
+		createTestCase('Available commands, prior command on the line', 'ls && c|', ['cd', 'code', 'code-insiders'], 'neither', availableSpecs, availableCommands);
+	});
+});
+
+function createTestCase(name: string, commandLineWithCursor: string, expectedSpecs: string[], resourcesRequested: 'files' | 'folders' | 'both' | 'neither', availableSpecs: Fig.Spec[], availableCommands: Set<string>): void {
```

Comment:
If you move this inside `suite` you'll be able to access variables you declare in that block. This lets you do things like this:

```
let commands: string[]

setup(() => {
  commands = ['default', 'set', ...];
});

function assertCompletions() {
  // references commands
}

test(() => {
  commands = ['custom test commands'];
});
```